### ISSUE TITLE
i broke a page...

### ISSUE THREAD

**Issue by dhwang90** [nert-nlp/Xposition#145]
http://flat.nert.georgetown.edu/ko/%EB%B6%80%ED%84%B0/Locus--Source/

I think that page has too many errors. I shouldnt have saved in the state it was in, but I did, and now the page isn't loading. Getting a HTTP ERROR 504.

Hewp.

**Comment by nschneid** [comment]
It's a bug in the `gex` macro renderer—apparently an infinite loop. https://github.com/nert-gu/Xposition/blob/8fc1ee0e17156e5a71ce018c3ee6cc91426c0075/wiki/plugins/macros/mdx/macro.py#L233

It occurs when the macro is called with `sent = '{일단은 여기[p ko/부터 Locus--Source] 갑니다.'` and `sent_gloss = ''`. It keeps putting empty strings into `columns`.

@ablodge can you look into this?

@dhwang90 in the meantime I've set it up to display an error when this issue is encountered. Other pages should work normally, so you can keep working on them.

**Comment by nschneid** [comment]
https://github.com/nert-gu/Xposition/blob/8fc1ee0e17156e5a71ce018c3ee6cc91426c0075/wiki/plugins/macros/mdx/macro.py#L247

I think if `end` is set to `0`, it will loop infinitely.

**Comment by nschneid** [comment]
The following solves some cases where there is a mismatched `{` or `}`:

```py
if word_gloss.strip():
    column = '<div class="gll">' + word_gloss.strip().replace('}', '&#125;') + '</div>'
elif end==0:
    column = sent.replace('{', '&#123;').replace('}', '&#125;')
    sent = ''
else:
    column = ''
```

But I'm able to break the macro in other ways, e.g. by omitting the sentence ID or putting `{{` where `{` is expected.

**Comment by ablodge** [comment]
A good behavior would be to do a try catch and if there’s any kind of error, then the inter-gloss shouldn’t display (rather than breaking the entire page).

**Comment by nschneid** [comment]
should be fixed by #192, which performs general error handling in macros
